algebra-utils/benches/fft/fft_bn382.rs

```diff
@@ -0,0 +1,386 @@
+#[macro_use]
+extern crate criterion;
+
+#[macro_use]
+extern crate bench_utils;
+
+use algebra::{
+    fields::bn_382::Fr,
+    PrimeField, UniformRand,
+};
+use algebra_utils::fft::{DensePolynomial, EvaluationDomain, BasicRadix2Domain, get_best_evaluation_domain};
+use rand;
+
+use std::{
+    fs::File,
+    path::Path,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use criterion::{BatchSize, BenchmarkId, Criterion};
+
+const DATA_PATH: &'static str = "./coeffs_bn382";
+
+fn save_data<F: PrimeField>(num_coeffs: usize) {
+
+    let mut fs = File::create(DATA_PATH).unwrap();
+    let rng = &mut rand::thread_rng();
+
+    for _ in 0..num_coeffs {
+        let elem:F = UniformRand::rand(rng);
+        match elem.write(&mut fs) {
+            Ok(_) => {},
+            Err(msg) => { panic!("Cannot save coeffs to file: {}", msg)}
+        }
+    }
+}
+
+fn load_data<F: PrimeField>(samples: usize) -> Vec<F> {
+    
+    if !Path::new(DATA_PATH).exists() {
+        save_data::<F>(1 << 23);
+    }
+    
+    let mut fs = File::open(DATA_PATH).unwrap();
+    let mut a: Vec<F> = Vec::with_capacity(samples);
+
+    for _i in 0..samples {
+        let elem1 = F::read(&mut fs).unwrap();
+        a.push(elem1);
+    }
+    a
+}
+
+fn bench_ffts<F: PrimeField, D: EvaluationDomain<F>>(
+    c: &mut Criterion,
+    num_coeffs: usize,
+    name: &'static str,
+) {
+    let mut group = c.benchmark_group(name);
+
+    // We expect the num_coeffs input to be a compatible size for the domain.
+    let domain = get_best_evaluation_domain::<F>(num_coeffs).unwrap();
+    let domain_size = domain.size();
+    assert_eq!(num_coeffs, domain_size);
+
+    group.bench_with_input(
+        BenchmarkId::from_parameter(num_coeffs),
+        &num_coeffs,
+        |b, _samples| {
+            b.iter_batched(
+                || {
+                    let a: Vec<F> = load_data(num_coeffs);
```

Comment:
I wonder why the FFT benches come with the load_data() and save_data() functions. Unlike generating points on an elliptic curve, sampling field elements is fast.